pkg/client/unversioned/clientcmd/loader.go

```diff
@@ -332,7 +332,17 @@ func LoadFromFile(filename string) (*clientcmdapi.Config, error) {
 	// set LocationOfOrigin on every Cluster, User, and Context
 	for key, obj := range config.AuthInfos {
 		obj.LocationOfOrigin = filename
+		if len(obj.Token) == 0 && len(obj.TokenFile) > 0 {
```

Comment:
I don't think we want to do this.  The reason is that code is executed after the merged config.  Combined with the precedence rules this can lead to unexpected behavior.  Here's an example:

Config-1:
```json
{
   "tokenFile": "someFile"
}
```

Config-2:
```json
{
   "token": "foo"
}
```

Now after merging, the combined authInfo is:

```json
{
   "tokenFile": "someFile",
   "token": "foo"
}
```

by precedence rules the token will be "foo" while it should actually be the contents of "someFile"

So I think we need to do the token loading pre-merge.

(although the merge logic is a little opaque to me, so I could be wrong)